internal/controller/utils/matching.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2025 Dynatrace LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package utils
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+func MatchContainerName(regex string, containerName string) (bool, error) {
```

Comment:
[CLARITY] It can be other things than `regex`
```suggestion
func MatchContainerName(pattern string, containerName string) (bool, error) {
```